packages/TermsAndConditions/Footnote/Footnote.jsx

```diff
@@ -0,0 +1,303 @@
+    transform: 'translateZ(0)',
+    backgroundColor: colorAthensGrey,
+    ...media.from('md').css({
+      maxHeight: 'calc(50vh - 57px)',
+    }),
+  },
+  ({ bodyHeight, isTextVisible }) => ({
+    height: bodyHeight,
+    opacity: isTextVisible ? 1 : 0,
+  })
+)
+
+const StyledListContainer = styled.div({
+  paddingTop: '1.5rem',
+  paddingBottom: '3rem',
+  ...media.from('md').css({
+    paddingTop: '0rem',
+    paddingBottom: '3rem',
+  }),
+})
+
+const FocusTrap = withFocusTrap('div')
+
+const usePrevious = value => {
+  const ref = useRef()
+  useEffect(() => {
+    ref.current = value
+  })
+  return ref.current
+}
+
+const Footnote = props => {
+  const { copy, number, content, returnRef, onClose, isOpen } = props
+  const closeRef = useRef(null)
+  const footnoteRef = useRef(null)
+  const headerRef = useRef(null)
+  const listRef = useRef(null)
+  const [data, setData] = useState({ content: null, number: null })
+  const [bodyHeight, setBodyHeight] = useState('auto')
+  const [isTextVisible, setIsTextVisible] = useState(true)
+
+  if ((content || number) && returnRef === null) {
+    warn('Footnote', 'A returnRef must be provided')
+  }
+
+  const prevProps = usePrevious(props)
+
+  const closeFootnote = e => {
+    returnRef.current.focus()
+    onClose(e)
+  }
+
+  // listen for ESCAPE, close button clicks, and clicks outside of the Footnote. Returns focus to returnRef and call onCloseClick
+  const handleClose = e => {
+    if (e.type === 'keydown') {
+      const key = e.keyCode || e.key
+      if (key === 'Escape' || key === 27) {
+        closeFootnote(e)
+      }
+    } else if (e.type === 'click') {
+      closeFootnote(e)
+    } else if (e.type === 'mousedown' && footnoteRef && !footnoteRef.current.contains(e.target)) {
+      closeFootnote(e)
+    }
+  }
+
+  const changeHeight = () => {
+    const oldHeight = listRef.current.offsetHeight
+    setBodyHeight(oldHeight)
+    setIsTextVisible(false)
+  }
+
+  const handleTransitionEnd = async e => {
+    e.persist()
+    if (e.propertyName === 'opacity' && !isTextVisible) {
+      await setData({ content, number })
+      const halfPageHeight = window.innerHeight * 0.5 - 57
+      const newHeight = listRef.current.offsetHeight
+      await setBodyHeight(newHeight > halfPageHeight ? halfPageHeight : newHeight)
+    }
+
+    if (e.propertyName === 'height' && !isTextVisible) {
+      setIsTextVisible(true)
+    }
+  }
+
+  // focus the close button on mount
+  useEffect(() => {
+    if (isOpen && closeRef && closeRef.current !== null) {
+      closeRef.current.focus()
+    }
+  }, [isOpen])
+
+  // add listeners for mouse clicks outside of Footnote and for ESCAPE key presses
+  useEffect(() => {
+    if (isOpen) {
+      window.addEventListener('mousedown', handleClose)
+      window.addEventListener('keydown', handleClose)
+    }
+    return () => {
+      if (isOpen) {
+        window.removeEventListener('mousedown', handleClose)
```

Comment:
This should be a `click` listener. Having it set to `mousedown` causes the panel to seemingly dismiss prematurely because a click has not yet been completed.